Orchestration/OrchestrationService/ModuleConfigurationDeployment.ps1

```diff
@@ -3086,28 +3209,42 @@ Function Get-ValidationResourceGroup() {
     param(
         [Parameter(Mandatory=$true)]
         [string]
-        $ArchetypeInstanceName
+        $ArchetypeInstanceName,
+        [Parameter(Mandatory=$false)]
+        [object]
+        $ValidationResourceGroupInformation
     )
 
     $resourceGroupName = `
-        Get-ValidationResourceGroupName `
-            -ArchetypeInstanceName $ArchetypeInstanceName;
+        (Get-ValidationResourceGroupConfiguration `
+            -ArchetypeInstanceName $ArchetypeInstanceName `
+            -ValidationResourceGroupInformation $ValidationResourceGroupInformation).Name;
 
     return `
         Get-AzResourceGroup $resourceGroupName `
             -ErrorAction SilentlyContinue;
 }
 
-Function Get-ValidationResourceGroupName() {
+Function Get-ValidationResourceGroupConfiguration() {
     [CmdletBinding()]
     param(
         [Parameter(Mandatory=$true)]
         [string]
-        $ArchetypeInstanceName
+        $ArchetypeInstanceName,
+        [Parameter(Mandatory=$false)]
+        [object]
+        $ValidationResourceGroupInformation
     )
 
-    return `
-        Get-UniqueString($ArchetypeInstanceName);
+    if($null -ne $ValidationResourceGroupInformation -and `
```

Comment:
I don't understand this condition, if the object is not null, and the name is not empty you return the object? else you return an object with just a name? Can you just return the string name?

Reply:
Added comments to clarify these conditions.